### DISCUSSION TITLE
Is it okay to have ~+10ms input latency in multiplayer?

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#25650] [Q&A]
It looks like I have input latency in multiplayer. It's much more comfortable to play in singleplayer. Everything is ok when I'm a single player in multiplayer room. The most sensitive part is keyboard - hit center aligns to right side on around 10ms average, when I'm trying to hit as usual according to rhythm. My singleplayer results have better accuracy. Also I really have some sort of discomfort in multiplayer games (everything is ok in stable osu multiplayer). Not sure what's wrong with it.

**Comment by peppy** [comment]
That sounds quiet weird, as nothing really changes in multiplayer. How are you seeing this latency?

**Reply by okolenmi** [reply]
Sorry for late answer. The problem was actual for me the whole time. I'm just was not sure as it's hard to talk about 10ms as humans too slow to sense it properly. Anyway, I have almost **the same** average hit error +-1ms for every try on the same map when a map is not difficult for me. So I tested it several times for some maps.

10ms offset shift will result around 5-7ms hit error - unintentionally I'm reading circles besides listening the song, so offset problems slightly fixed even with bad map offset.

After some tests in multiplayer I have an average **+6ms** hit error with perfect offset (+-1ms hit error  in singleplayer) on maps near 5.2* (usually have FC on such starrate). Hit error is around 3ms when map is around 4* - it looks like I pay more attention to approach circles on slow maps to have ability to play low AR with good accuracy.

I wonder if there any mechanics that can compress replay in real time to show it to other players. Maybe problem exist in one of such mechanics. As I say before, I have no broblems when I'm a single player host in multiplayer room.

After all my experiments with keyboards and drivers I have almost instant hitsound in singleplayer, but I can clearly feel a delay in multiplayer. 

My settings (144Hz refresh):
![изображение](https://github.com/ppy/osu/assets/55319814/88bf7972-9392-43a8-871d-948e6f3868e0)

**Reply by okolenmi** [reply]
It's not only me:

![изображение](https://github.com/ppy/osu/assets/55319814/86a9e69e-d2bd-4542-b1ce-a22e250ab904)

**Reply by okolenmi** [reply]
The lag remains even in singleplayer after playing for a ~long time in multiplayer. After restarting the game it disappears. It seems that the problem is not in the multiplayer itself but in something like a memory leak (too much resources loaded in the same time, for example), but it affects the delay more than FPS - fps is nearly 1000 the whole time.

**Comment by huyenden** [comment]
The delay is normal for me, try turning on multithreaded.
